cmd/oras/internal/display/metadata/template/manifest_fetch.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package template
+
+import (
+	"reflect"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras/cmd/oras/internal/display/metadata"
+	"oras.land/oras/cmd/oras/internal/display/metadata/model"
+)
+
+// ManifestFetchHandler handles JSON metadata output for manifest fetch events.
+type ManifestFetchHandler struct {
+	template string
+}
+
+// NewManifestFetchHandler creates a new handler for manifest fetch events.
+func NewManifestFetchHandler(template string) metadata.ManifestFetchHandler {
+	return &ManifestFetchHandler{
+		template: template,
+	}
+}
+
+// OnFetched is called after the manifest fetch is completed.
+func (ph *ManifestFetchHandler) OnFetched(manifest ocispec.Manifest) error {
+	return parseAndWrite(model.ToMappable(reflect.ValueOf(manifest)), ph.template)
```

Comment:
Can we use `json.Marshal` and the n`json.Unmarshal` instead of `ToMappable`? It is safer although slower.

Reply:
Maybe I am missing sth here, but how does the marshall+unmarshall operation convert all fields into a map?